### PR TITLE
Debt - 4457 - Remove talent paths

### LINKED ISSUE #4457: Review profile routes
## 👋 Introduction

We need to review the profile routes and attempt to remove any `/talent/*`  paths since they are being phased out. For more info, reference the discussion on https://github.com/GCTC-NTGC/gc-digital-talent/pull/4442#discussion_r1004537866

## ✅ Acceptance Criteria

- [ ] Deprecate all `/talent/*` routes
- [ ] Redirect them to the new versions
- [ ] Write tests to confirm redirects exist and work (❓)
- [ ] Ensure the `/talent/search` routes still Permanent Redirect to `/search` to accommodate managers given the old link in the past.

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

This deprecates `/talent/profile` and replaces it with `/users/me` while redirecting the new path to `/users/{userId}/profile`.

> **Note**
> This was requested to remove all instances of `/profile/*` before launching direct intake.

## 🧪 Testing

1. Build talent search `npm run production --workspace=talentsearch`
2. Navigate to `/en/users/me` and ensure it redirects to the profile
3.  Navigate to `/en/talent/profile` and ensure it redirects to the profile
4. Click the "My profile" link in the header and ensure it goes from `/en/users/me` -> `/users/{userId}/profile`

## 🤖 Robot Stuff

Resolves #4457

### CODE REVIEW

File: frontend/cypress/integration/talentsearch/redirects.spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { aliasQuery } from "../../support/graphql-test-utils";
+
+describe("Redirects", () => {
+
+
+
+  const expectToBeOnProfile = () => {
+    cy.wait("@gqlgetMeQuery");
+    cy.url()
+    .should('match', /users\/[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}\/profile$/g);
```
Comment (petertgiles) [inline frontend/cypress/integration/talentsearch/redirects.spec.js:10]: Might be nice to have the regex for a UUID a named constant somewhere to make it clear what this insanity represents and allow it to be reused. :grin:

**Review by petertgiles — APPROVED** [review]
Very nice! :dart:   One small suggestion below.

File: frontend/cypress/integration/talentsearch/redirects.spec.js (hunk 1 above)
Comment (esizer) [reply frontend/cypress/integration/talentsearch/redirects.spec.js:10]: Oh, I wanted to do that but i don't know how to do that with regex! 😓

Comment (esizer) [reply frontend/cypress/integration/talentsearch/redirects.spec.js]: Thanks! Added in [move uuid regex to variable for more clarity](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4460/commits/1bd8aed6a5a3f93f5988c742c6361b15f42b07f2)

**Pull request merged**